pkg/cmd/server/admission/init.go

```diff
@@ -22,6 +23,7 @@ type PluginInitializer struct {
 	RESTClientConfig      restclient.Config
 	Informers             shared.InformerFactory
 	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
+	DefaultRegistryFn     imageapi.DefaultRegistryFunc
```

Comment:
No, this is right, because this is what image streams use to detect
registry.  The other code is wrong.

On Fri, Aug 5, 2016 at 8:25 AM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/server/admission/init.go
> <https://github.com/openshift/origin/pull/8995#discussion_r73683918>:
>
> > @@ -22,6 +23,7 @@ type PluginInitializer struct {
> >  	RESTClientConfig      restclient.Config
> >  	Informers             shared.InformerFactory
> >  	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
> > +	DefaultRegistryFn     imageapi.DefaultRegistryFunc
>
> Didn't know this already existed. It's wrong. There are N aliases for the
> registry and the admission plugin must recognize all of them. There should
> be a separate function for "get me the default name to use for making
> references in the cluster", but there are definitely cases where you need
> all the aliases and those aliases can change.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/f96f5e56ba70c7c0f24f875621efdf1ef45ce3c0#r73683918>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6DI81Q_uN-ilGLLMCRG2Xrgeg4Eks5qcyvWgaJpZM4IlDlb>
> .
>